### DISCUSSION TITLE
Referral link features

### DISCUSSION THREAD

**Discussion by dbwebapp** [BitCannaGlobal/WeedWallet#78] [Ideas]
We propose the development of a new and relevant tool:

For validators, attracting new users to stake their coins on their node is crucial for long-term survival. That's why we believe that the delegation process should be made easier and more streamlined for users.

Let's take an example. Some validators are highly active on Discord, providing assistance to users in understanding project concepts and addressing technical queries. When a solution is delivered and they invest time in answering their questions, people are usually very grateful. Our proposal is to have a pre-built link, similar to a referral link, that the user receiving help can click on. This link would redirect them to a dedicated page within the WeedWallet where they can seamlessly go through the delegation process.

The streamlined process would involve the following steps on the dedicated page:

First window: Connect wallet
Second window: Select the amount to delegate (if available). If not, a new window appears suggesting a link to Osmosis for purchasing coins or a button to redelegate to the validator connected with the referral link.
Clicking on the delegate button will automatically delegate to the Validator's Referral link. The process incurs no cost for the user to redelegate or delegate if they have coins available in their wallet. This approach helps educate users about delegation and serves as an act of gratefulness for receiving help from a validator.

Currently, if a user wishes to collaborate and delegate, they have to go through a more cumbersome process, which includes:

Opening ping.pub, Restake, or Weedwallet platform
Connecting their wallet
Clicking on the validators button
Searching for the validator (often requiring a return to Discord to check the validator's name)
Selecting the validator
Clicking on the delegate button
Entering the delegation amount
Confirming the delegation.
If we can reduce this entire process to just two simple steps:

-Connect wallet
- Enter the amount to delegate and press Delegate button (to the validator associated with the referral link clicked)
That's it! This streamlined approach would greatly enhance the user experience and make the delegation process much more convenient and user-friendly.

**Comment by TJ-BitCanna** [comment]
Did you notice it is possible already?

Test for yourself and let me know if it is what you are looking for. Make sure to log out from the WeedWallet first to test the full experience you seek :)

https://wallet-testnet.bitcanna.io/validators/bcnavaloper1fvl05kvcn6y7292xvt2cy8jwcgu92j34j9d3qe

**Comment by dbwebapp** [comment]
Yes, we could go one step further by having this folllowing prompt directly from clicking the link  :
![Screenshot 2023-05-25 at 10 59 51](https://github.com/BitCannaGlobal/WeedWallet/assets/96944913/9b41bb2d-ba59-447c-bdbe-3b9b82209e09)
If wallet not connected instead of NEXT STEP BUTTON , we could have CONNECT wallet , so user are not going away from this prompt

Also i tried to delegate from there and buttons 1/4 1/2 all not working , just 0 appears on the input  :  I need to enter manually the number 
![Screenshot 2023-05-25 at 10 57 46](https://github.com/BitCannaGlobal/WeedWallet/assets/96944913/c09ff863-a463-4792-9221-7dc4b72031b2)

**Comment by TJ-BitCanna** [comment]
Yes, but to get that prompt immediately it would need to be placed on a separate URL to be able to differentiate between the various use cases. Because in the past we also talked about having a direct URL to the validator page (without the prompt :)).

So you have to be able to differentiate between having a link with and without the prompt; which is effectively only done with a different URL. For now my preference would be to "just" offer the direct link to the validator page; finding the delegate button should be an easy task for an user.

We'll check on the buttons though.

**Comment by TJ-BitCanna** [comment]
Suggested issue:
- check the code behind the "1/4", "1/2", "max" options on the pop-up when delegating to a validator

**Comment by TJ-BitCanna** [comment]
The validator page has been gone since the update, but we will bring it back shortly.
